lua/rustaceanvim/lsp/init.lua

```diff
@@ -93,6 +94,45 @@ local function configure_file_watcher(server_cfg)
   end
 end
 
+---LSP restart ininer implementations
+---@param bufnr? number
+---@param set_target_callback? function(client) Optional callback to run for each client before restarting.
+---@return number|nil client_id
+local function restart(bufnr, set_target_callback)
```

Comment:
quibble: I would just call this `callback`, as this function doesn't know what it does.

Reply:
done